src/compactor.rs

```diff
@@ -135,7 +135,10 @@ impl CompactorOrchestrator {
 
     fn run(&mut self) {
         let ticker = crossbeam_channel::tick(self.options.poll_interval);
-        loop {
+
+        // Stop the loop when the executor is shut down *and* all remaining
+        // `worker_rx` messages have been drained.
+        while !(self.executor.is_stopped() && self.worker_rx.is_empty()) {
             crossbeam_channel::select! {
                 recv(ticker) -> _ => {
                     self.load_manifest().expect("fatal error loading manifest");
```

Comment:
shouldn't `load_manifest()` check for `is_stopped()`?

Reply:
I think it's not the end of the world if we do. All that will happen is that we may refresh the manifest and then try to schedule a compaction that the executor rejects.

Reply:
Ah. I didn't think of this. I don't mind fixing it for completeness's sake. Question is where? I initially thought of doing:

```rust
    fn load_manifest(&mut self) -> Result<(), SlateDBError> {
        self.tokio_handle.block_on(self.manifest.refresh())?;
        if !self.executor.is_stopped() {
            self.refresh_db_state()?;
        }
        Ok(())
    }
```

But load_manifest is called from `finish_compaction` which can run after the executor is stopped. Should I put the `is_stopped` check in the recv(ticker) block? Since it's not async, I think it should be OK.

Reply:
why does it matter that it's not async?

Reply:
See my comment below. I could be misunderstanding how async works.